otcextensions/sdk/cloud/cce.py

```diff
@@ -199,3 +199,216 @@ def delete_cce_cluster(
+        spec['login']['sshKey'] = keypair
+        spec['rootVolume']['volumetype'] = root_volume_type.upper()
+        if root_volume_size < 40:
+            raise ValueError('Root volume size %s is lower than 40 GB.'
+                             % root_volume_size)
+        spec['rootVolume']['size'] = root_volume_size
+
+        for item in data_volumes:
+            for key in item:
+                if key.upper() not in volume_types:
+                    raise ValueError('data volume type %s must be one of '
+                                     'the following choices: %s'
+                                     % key, volume_types)
+                if not (100 <= item[key] <= 32768):
+                    raise ValueError('The data volume size must be specified '
+                                     'between 100 and 32768 GB.')
+                spec['dataVolumes'].append({
+                    'volumetype': key.upper(),
+                    'size': item[key]
+                })
+
+        if availability_zone:
+            spec['az'] = availability_zone
+        if count:
+            spec['count'] = count
+        if dedicated_host:
+            spec['dedicatedHostId'] = dedicated_host
+        if ecs_group:
+            spec['ecs_group'] = ecs_group
+        if fault_domain:
+            spec['faultDomain'] = fault_domain
+        if floating_ip:
+            spec['publicIP'] = floating_ip
+        if k8s_tags:
+            spec['k8sTags'] = k8s_tags
+        if lvm_config:
+            spec['extendParam']['DockerLVMConfigOverride'] = lvm_config
+        if max_pods:
+            spec['extendParam']['maxPods'] = max_pods
+        if node_image_id:
+            spec['extendParam']['alpha.cce/NodeImageID'] = node_image_id
+        if offload_node:
+            spec['offloadNode'] = offload_node
+        if os:
+            spec['os'] = os
+        if postinstall_script:
+            spec['extendParam']['alpha.cce/preInstall'] = postinstall_script
+        if preinstall_script:
+            spec['extendParam']['alpha.cce/preInstall'] = preinstall_script
+        if tags:
+            spec['userTags'] = tags
+
+        cluster = self.cce.find_cluster(
+            name_or_id=cce_cluster,
+            ignore_missing=True)
+        if not cluster:
+            raise ReferenceError('Cluster %s not found.' % cce_cluster)
+        obj = self.cce.create_cluster_node(
+            cluster=cluster.id,
+            metadata=metadata,
+            spec=spec
+        )
+
+        if obj.job_id and wait:
+            wait_args = {}
+            if wait_interval:
+                wait_args['interval'] = wait_interval
+            if wait_timeout:
+                wait_args['wait'] = wait_timeout
+
+            self.cce.wait_for_job(obj.job_id, **wait_args)
+            obj = self.cce.get_cluster(obj.id)
+
+        return obj
+
+    def delete_cce_cluster_node(
+        self,
+        wait=True, wait_timeout=180, wait_interval=5,
```

Comment:
I suppose, that 3mins in `wait_timeout` is very little time

Reply:
It ran through every, so I did not suppose to increase the time right now. Deletion of the node is fast. Deletion of the cluster takes a lot of time.

Reply:
increased to 300